tests/suites/test_suite_psa_crypto_se_driver_hal.function

```diff
@@ -720,7 +733,7 @@ static int smoke_test_key(mbedtls_svc_key_id_t key)
                                         buffer, sizeof(buffer), NULL, 0,
                                         buffer, sizeof(buffer), &length));
 
-#if defined(PSA_WANT_ALG_SHA_256)
+#if defined(PSA_WANT_ALG_SHA_256) & defined(MBEDTLS_PSA_BUILTIN_ALG_HKDF)
```

Comment:
I think we should use `&&` here for consistency. I don't think `&` is incorrect, but it's surprising. I can't seem to push an update, so I'll create a follow-up.